### PR TITLE
Add WhittleFactory::buildWithCriteria

### PR DESCRIPTION
**Pull request by jschueller**
- Allows to disambiguate whether to return information criteria
- Return it in a tuple instead of by reference
- Removed useless ARMAFactoryImplementation::build with reference

### CODE REVIEW

**Comment by sofianehaddad** [comment]
@jschueller agreed that the changes allow "to disambiguate"
But IMHO in order to have an "homogeneous code" if changes are done here, we should have the same kind of **buildWithXXX* methods for all ones that had a Point ref. What do you think ?

**Comment by jschueller** [comment]
That's the idea, are there remaining others ?

File: lib/src/Uncertainty/Process/ARMAFactoryImplementation.cxx [hunk 1]
```diff
@@ -155,24 +155,12 @@ ARMA ARMAFactoryImplementation::build(const TimeSeries & timeSeries) const
...
 
-ARMA ARMAFactoryImplementation::build(const TimeSeries & timeSeries,
-                                      Point & informationCriteria) const
-{
-  throw NotYetImplementedException(HERE) << "In ARMAFactoryImplementation::build(const TimeSeries & timeSeries, Point & informationCriteria) const";
-}
-
 /* Build method */
 ARMA ARMAFactoryImplementation::build(const ProcessSample & sample) const
 {
```
Comment (sofianehaddad) [inline lib/src/Uncertainty/Process/ARMAFactoryImplementation.cxx]: I think we should keep only ARMAFactoryImplementation::build methods : 

  *ARMAFactoryImplementation::build(const TimeSeries & timeSeries,  Point & informationCriteria) const*

  *ARMAFactoryImplementation::build(const ProcessSample & sample,  Point & informationCriteria) const*

Both **ARMAFactoryImplementation::build(const TimeSeries & timeSeries) const** and 
  **ARMAFactoryImplementation::build(const ProcessSample & sample) const** should be removed

What do you think ?

Comment (jschueller) [inline lib/src/Uncertainty/Process/ARMAFactoryImplementation.cxx]: I thought returning a Point for the criteria is specific to WhittleFactory criteria, so I ditched them. Anyways it's not available from ARMAFactory interface. But as there are no other ARMAFactoryImplementation other than WhittleFactory it's hard to tell.

Comment (sofianehaddad) [inline lib/src/Uncertainty/Process/ARMAFactoryImplementation.cxx]: ok

**Comment by regislebrun** [comment]
+1

**Comment by jschueller** [comment]
tests fails, related to hmat: https://github.com/jeromerobert/hmat-oss/pull/51

**Pull request merged**